pkg/pool-weighted/test/foundry/ManagedPoolTokenLib.t.sol

```diff
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+
+import { Test } from "forge-std/Test.sol";
+
+import "@balancer-labs/v2-interfaces/contracts/solidity-utils/openzeppelin/IERC20.sol";
+import "@balancer-labs/v2-solidity-utils/contracts/test/TestToken.sol";
+
+import "../../contracts/WeightedMath.sol";
+import "../../contracts/lib/ValueCompression.sol";
+
+import "../../contracts/test/MockManagedPoolTokenLib.sol";
+import "../../contracts/managed/ManagedPoolTokenLib.sol";
+
+contract ManagedPoolTokenLibTest is Test {
+    uint256 private constant _START_DENORM_WEIGHT_OFFSET = 0;
+    uint256 private constant _END_DENORM_WEIGHT_OFFSET = _START_DENORM_WEIGHT_OFFSET + _DENORM_WEIGHT_WIDTH;
+    uint256 private constant _DECIMAL_DIFF_OFFSET = _END_DENORM_WEIGHT_OFFSET + _DENORM_WEIGHT_WIDTH;
+
+    uint256 private constant _DENORM_WEIGHT_WIDTH = 64;
+    uint256 private constant _DECIMAL_DIFF_WIDTH = 5;
+
+    uint256 private constant _MAX_DENORM_WEIGHT = 1e22;
+
+    MockManagedPoolTokenLib private mock;
+
+    function setUp() external {
+        mock = new MockManagedPoolTokenLib();
+    }
+
+    function clearWordAtPosition(
+        bytes32 word,
+        uint256 offset,
+        uint256 bitLength
+    ) internal returns (bytes32 clearedWord) {
+        uint256 mask = (1 << bitLength) - 1;
+        clearedWord = bytes32(uint256(word) & ~(mask << offset));
+    }
+
+    function assertOtherStateUnchanged(
+        bytes32 oldPoolState,
+        bytes32 newPoolState,
+        uint256 offset,
+        uint256 bitLength
+    ) internal returns (bool) {
+        bytes32 clearedOldState = clearWordAtPosition(oldPoolState, offset, bitLength);
+        bytes32 clearedNewState = clearWordAtPosition(newPoolState, offset, bitLength);
+        assertEq(clearedOldState, clearedNewState);
+    }
+
+    function testScalingFactor(bytes32 tokenState, uint8 decimals) external {
+        ERC20 token = new TestToken("Test", "TEST", decimals);
```

Comment:
```suggestion
        vm.assume(decimals <= 30);
        ERC20 token = new TestToken("Test", "TEST", decimals);
```

Just to avoid having too many tests be rejected due to too many decimals.